addons/account_audit_trail/models/mail_message.py

```diff
@@ -43,32 +69,116 @@ def _compute_account_audit_log_preview(self):
+            '|', ('company_id', '=', False), ('company_id.check_account_audit_trail', '=', True),
+            '|', ('customer_rank', '>', 0), ('supplier_rank', '>', 0),
+        ])
+
+    def _search_account_audit_log_partner_id(self, operator, value):
+        return self._search_audit_log_related_record_id('res.partner', operator, value)
+
+    def _compute_account_audit_log_display_name(self):
+        for message in self:
+            message.account_audit_log_display_name = (
+                message.account_audit_log_move_id
+                or message.account_audit_log_account_id
+                or message.account_audit_log_tax_id
+                or message.account_audit_log_partner_id
+                or message.account_audit_log_company_id
+            ).display_name
+
+    def _compute_show_audit_log(self):
+        for message in self:
+            message.show_audit_log = message.message_type == 'notification' and (
+                message.account_audit_log_move_id
+                or message.account_audit_log_account_id
+                or message.account_audit_log_tax_id
+                or message.account_audit_log_partner_id
+                or message.account_audit_log_company_id
+            )
+
+    def _search_show_audit_log(self, operator, value):
+        if operator not in ['=', '!='] or not isinstance(value, bool):
+            raise UserError(_('Operation not supported'))
+        return [('message_type', '=', 'notification')] + OR([
+            [('model', '=', 'account.move'), ('res_id', 'in', self.env['account.move']._search([
+                ('company_id.check_account_audit_trail', operator, value),
+            ]))],
+            [('model', '=', 'account.account'), ('res_id', 'in', self.env['account.account']._search([
+                ('company_id.check_account_audit_trail', operator, value),
+            ]))],
+            [('model', '=', 'account.tax'), ('res_id', 'in', self.env['account.tax']._search([
+                ('company_id.check_account_audit_trail', operator, value),
+            ]))],
+            [('model', '=', 'res.partner'), ('res_id', 'in', self.env['res.partner']._search([
+                '|', ('company_id', '=', False), ('company_id.check_account_audit_trail', operator, value),
+                '|', ('customer_rank', '>', 0), ('supplier_rank', '>', 0),
+            ]))],
+            [('model', '=', 'res.company'), ('res_id', 'in', self.env['res.company']._search([
+                ('check_account_audit_trail', operator, value),
+            ]))],
+        ])
+
+    def _compute_audit_log_related_record_id(self, model, fname, domain):
+        messages_of_related = self.filtered(lambda m: m.model == model and m.res_id)
+        (self - messages_of_related)[fname] = False
+        if messages_of_related:
+            related_recs = self.env[model].sudo().search([('id', 'in', messages_of_related.mapped('res_id'))] + domain)
```

Comment:
Shouldn't we use `expression.AND` ? I mean your solution is working but isn't it better to use the `expression.AND` here?

Reply:
It would be better if I did `domain + [('id', 'in', messages_of_related.mapped('res_id'))]` because then there would be a (very unlikely) domain injection possibility, but adding `expression.AND` in this case just adds some useless checks :shrug: